src/shogun/lib/Map.h

```diff
@@ -292,7 +292,8 @@ IGNORE_IN_CLASSLIST template<class K, class T> class CMap: public CSGObject
 	 */
 	int32_t hash(const K& key)
 	{
-		return CHash::MurmurHash3((uint8_t*)(&key), sizeof(key), 0xDEADBEEF) % hash_size;
+		int32_t hv=get_hash_value(key)% (uint32_t) hash_size;
```

Comment:
@vigsterkr 
The unit test will fail if int32_t `get_hash_value(key)%  hash_size;`